kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/InterfaceState.java

```diff
@@ -17,38 +17,52 @@
 import org.eclipse.kura.net.ConnectionInfo;
 import org.eclipse.kura.net.IPAddress;
 import org.eclipse.kura.net.NetInterfaceType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class InterfaceState {
 
-    private final String m_name;
-    private final boolean m_up;
-    protected boolean m_link;
-    private final IPAddress m_ipAddress;
+    private static final Logger logger = LoggerFactory.getLogger(InterfaceState.class);
+    private final String name;
+    private final boolean up;
+    protected boolean link;
+    private final IPAddress ipAddress;
 
     public InterfaceState(String interfaceName, boolean up, boolean link, IPAddress ipAddress) {
-        this.m_name = interfaceName;
-        this.m_up = up;
-        this.m_link = link;
-        this.m_ipAddress = ipAddress;
+        this.name = interfaceName;
+        this.up = up;
+        this.link = link;
+        this.ipAddress = ipAddress;
     }
 
     public InterfaceState(NetInterfaceType type, String interfaceName) throws KuraException {
-        this.m_name = interfaceName;
-        this.m_up = LinuxNetworkUtil.hasAddress(interfaceName);
-        this.m_link = LinuxNetworkUtil.isLinkUp(type, interfaceName);
+        this.name = interfaceName;
+        this.up = LinuxNetworkUtil.hasAddress(interfaceName);
+        this.link = LinuxNetworkUtil.isLinkUp(type, interfaceName);
+        logger.debug("InterfaceState() :: {} - link?={}", interfaceName, this.link);
+        logger.debug("InterfaceState() :: {} - up?={}", interfaceName, this.up);
+        ConnectionInfo connInfo = new ConnectionInfoImpl(interfaceName);
+        this.ipAddress = connInfo.getIpAddress();
+    }
 
+    public InterfaceState(NetInterfaceType type, String interfaceName, boolean isL2OnlyInterface) throws KuraException {
```

Comment:
Doesn't this change imply an update of the exported package in manifest file?

Reply:
Made changes to the MANIFEST.MF